src/deploy/db/mongo-adapter/mongo-adapter.ts

```diff
@@ -119,10 +125,10 @@ export class MongoDBAdapter {
     if (version) {
       finalVersion = version;
 
-      if (version !== deployedV || !deployedV) {
+      if (version !== deployedV?.dbVersion || !deployedV) {
         // we should only have a single TEMP version at any given time
-        if (version !== tempV && tempV) {
-          await this.clearDBForVersion(tempV);
+        if (version !== tempV?.dbVersion && tempV) {
```

Comment:
same as above, you can reverse the order of the if statement and remove the use of the `?`